File: app/src/main/java/core/legion/samovar/data/FirebaseManagerModule.java
package core.legion.samovar.data;

import android.content.Context;

import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import com.google.firebase.firestore.FirebaseFirestore;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class FirebaseManagerModule {

    @Provides
    static FirebaseOptions firebaseOptions(Context context) {
        return FirebaseOptions.fromResource(context);
    }

    @Provides
    static FirebaseApp firebaseApp(Context context, FirebaseOptions firebaseOptions) {
        return FirebaseApp.getInstance();
//        return FirebaseApp.initializeApp(context, firebaseOptions);
    }

    @Provides
    static FirebaseFirestore firebaseFirestore(FirebaseApp firebaseApp) {
        return FirebaseFirestore.getInstance(firebaseApp);
    }

    @Binds
    abstract DBManager firebaseManager(FirebaseManager manager);
}

File: app/src/main/java/core/legion/samovar/data/FirebaseManagerModule.kt
package core.legion.samovar.data

import com.google.firebase.FirebaseApp
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.storage.FirebaseStorage
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class FirebaseManagerModule {

    @Binds
    abstract fun firebaseManager(manager: FirebaseManager): DBManager

    @Module
    companion object {

        @JvmStatic
        @Provides
        fun firebaseApp(): FirebaseApp {
            return FirebaseApp.getInstance()!!
        }

        @JvmStatic
        @Provides
        fun firebaseFirestore(firebaseApp: FirebaseApp): FirebaseFirestore {
            return FirebaseFirestore.getInstance(firebaseApp)
        }

        @JvmStatic
        @Provides
        fun firebaseStorage(firebaseApp: FirebaseApp): FirebaseStorage {
            return FirebaseStorage.getInstance(firebaseApp)
        }
    }
}
